play-json/shared/src/main/scala/Json.scala

```diff
@@ -414,6 +419,29 @@ object Json extends JsonFacade {
   sealed trait MacroOptions
 
   /**
+   * Defines the default macro options.
+   *
+   * Since you can't have defaults for a type parameter (unless it's a contravariant type, it will default to
+   * Nothing), we supply the default via an implicit parameter.
+   */
+  trait DefaultMacroOptions[O <: Json.MacroOptions]
+
+  /**
+   * Low priority implicit used when Json.MacroOptions is not passed.
```

Comment:
We don't need evidence that the options conform to the option type, the type bounds already do that. If I invoke `JsonConfiguration[MyOptions]()`, the compiler is able to do all the necessary type checking and passing of types without this type class, without any evidence of anything. But, if I invoke `JsonConfiguration()`, then in the absence of any type class supplying evidence, `Nothing` is passed as the type. That's why this typeclass is needed, it supplies the evidence for what the default options should be when none is selected. If it weren't for that one use case, we could remove it.